WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java

```diff
@@ -474,41 +483,48 @@ private void approveComment() {
-                @Override
-                public void onActionResult(CommentActionResult result) {
-                    if (result != null && result.isSuccess()) {
-                        requestCompleted(ARG_ACTION_APPROVE);
-                    } else {
-                        requestFailed(ARG_ACTION_APPROVE);
-                    }
-                }
-            });
+            // Update pseudo comment (built from the note)
+            CommentModel comment = mNote.buildComment();
+            comment.setStatus(CommentStatus.APPROVED.toString());
+            SiteModel site = mSiteStore.getSiteBySiteId(mNote.getSiteId());
+            if (site == null) {
+                AppLog.e(T.NOTIFS, "Impossible to approve a comment on a site that is not in the App. SiteId: "
+                                   + mNote.getSiteId());
+                requestFailed(ARG_ACTION_APPROVE);
+                return;
+            }
 
+            // Push the comment
+            mDispatcher.dispatch(CommentActionBuilder.newPushCommentAction(new RemoteCommentPayload(site, comment)));
         }
 
-        private void replyToComment(){
-            if (mNote == null) return;
+        private void replyToComment() {
+            if (mNote == null) {
+                return;
+            }
 
             // Bump analytics
             AnalyticsTracker.track(AnalyticsTracker.Stat.NOTIFICATION_QUICK_ACTIONS_REPLIED_TO);
 
-
             if (!TextUtils.isEmpty(mReplyText)) {
-                CommentActions.submitReplyToCommentNote(mNote, mReplyText, new CommentActions.CommentActionListener() {
-                    @Override
-                    public void onActionResult(CommentActionResult result) {
-                        if (result != null && result.isSuccess()) {
-                            requestCompleted(ARG_ACTION_REPLY);
-                        } else {
-                            if (result != null && !TextUtils.isEmpty(result.getMessage())) {
-                                requestFailedWithMessage(result.getMessage(), true);
-                            } else {
-                                requestFailed(ARG_ACTION_REPLY);
-                            }
-                        }
-                    }
-                });
+                SiteModel site = mSiteStore.getSiteBySiteId(mNote.getSiteId());
+                if (site == null) {
+                    AppLog.e(T.NOTIFS, "Impossible to reply a comment on a site that is not in the App. SiteId: "
+                                       + mNote.getSiteId());
+                    requestFailed(ARG_ACTION_APPROVE);
+                    return;
+                }
+
+                // Pseudo comment (built from the note)
+                CommentModel comment = mNote.buildComment();
+
+                // Pseudo comment reply
+                CommentModel reply = new CommentModel();
```

Comment:
Well, the thing is we don't want to store "draft" comment in the app, there is no such thing as a local only comment (We use PerstistentEditText to avoid re-typing a comment if the app force close or is dismissed). 

If the comment reply fails or if the app crash, the comment disappear but the "text" remains. That's why I never used `INSTANTIATE_COMMENT`, I want to avoid any "phantom" comments in the comment list (comment the user could see in the comment list but not on their site). 

I think this can be used in the future if we continue to work on the comment section, we could add a new status (local draft?) and use INSTANTIATE_COMMENT here.